sklearn/plot/_gridsearch_results.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+from sklearn.plot import plot_heatmap
+
+
+def _plot_1D_results(cv_results, params, ax, xlabel, ylabel, title,
+                     fmt, xtickrotation):
+    import matplotlib.pyplot as plt
```

Comment:
yea we could. but then this was being checked in plot_heatmap again for the 2D case. so i did separately for 1D case. To make sure it is always closer to the public layer, maybe we should make the `ax` parameter of plot_heatmap compulsory and whichever function (grid search or confusion matrix or any other in future) calls it should perform the ax is None check. works?